### PR TITLE
modify scripts for nightly runner

### PR DESCRIPTION
**Pull request by ewlu**

### CODE REVIEW

File: scripts/get_previous_timestamp.py [hunk 1]
```diff
@@ -35,24 +41,38 @@ def get_workflow_runs(token: str, repo: str):
...
+            f.write(runs[0]['id'])
+        else:
+            f.write(runs[1]['id'])
 
 def main():
     args = parse_arguments()
     runs = get_workflow_runs(args.token, args.repo)
+    if runs == None:
+        print("Server failure")
+        return
```
Comment (patrick-rivos) [inline scripts/get_previous_timestamp.py:71]: Should we throw an error here so the github ui shows an x?

File: scripts/get_previous_timestamp.py [hunk 2]
```diff
@@ -35,24 +41,38 @@ def get_workflow_runs(token: str, repo: str):
...
 
 def get_run_timestamp(runs, run_id: str):
-    runs = [run for run in runs if run["name"] == "Patchworks"]
     with open("date_cur.txt", "w") as f:
         if str(runs[0]['id']) != str(run_id):
             f.write(runs[0]['created_at'])
         else:
             f.write(runs[1]['created_at'])
 
+def get_run_id(runs, run_id: str)
```
Comment (patrick-rivos) [inline scripts/get_previous_timestamp.py:59]:
```suggestion
def write_run_id(runs, run_id: str)
```

File: scripts/get_previous_timestamp.py [hunk 3]
```diff
@@ -35,24 +41,38 @@ def get_workflow_runs(token: str, repo: str):
...
 def main():
     args = parse_arguments()
     runs = get_workflow_runs(args.token, args.repo)
+    if runs == None:
+        print("Server failure")
+        return
     with open("runs.log", "w") as f:
         f.write(json.dumps(runs['workflow_runs'][:10], indent=4))
     get_run_timestamp(runs['workflow_runs'], args.run_id)
+    get_run_id(runs['workflow_runs'], args.run_id)
```
Comment (patrick-rivos) [inline scripts/get_previous_timestamp.py:75]:
```suggestion
    write_run_id(runs['workflow_runs'], args.run_id)
```

File: scripts/get_previous_timestamp.py [hunk 4]
```diff
@@ -35,24 +41,38 @@ def get_workflow_runs(token: str, repo: str):
...
+    if r.status_code >= 500:
+        with open("patchwork_down.txt", "w") as f:
+            f.write(f"status code: {r.status_code}")
+        return None
+    runs = [run for run in json.loads(r.text) if run["name"] == workflow]
+    print(f"After filter have {len(runs)} to consider")
     return runs
 
 def get_run_timestamp(runs, run_id: str):
-    runs = [run for run in runs if run["name"] == "Patchworks"]
```
Comment (patrick-rivos) [inline scripts/get_previous_timestamp.py:42]: Why remove this?
https://github.com/patrick-rivos/riscv-gnu-toolchain/commit/74362c59f40fca31545c8359698af2829fe739d7
Presumably we don't want to count the nightly-run as an actual run?

File: scripts/check_patch_checks.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    )
+    return parser.parse_args()
+
+def make_api_request(url):
+    print(url)
+    r = requests.get(url)
+    return json.loads(r.text)
+
+def check_patch(patch):
+    url = patch['checks']
```
Comment (patrick-rivos) [inline scripts/check_patch_checks.py:32]:
```suggestion
    """ Check if the given patch has already been seen by the CI. """
    url = patch['checks']
```

File: scripts/check_patch_checks.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    # If testsuite was not reported, mark for rerun
+    # testsuite should be reported as long as it was applied
+    if not apply_failure and not testsuite_reported:
+        return True
+
+    return False
+
+
+def get_patches(start: str, end: str):
+    url = f"https://patchwork.sourceware.org/api/1.3/patches/?order=date&q=RISC-V&project=6&since={start}&before={end}"
```
Comment (patrick-rivos) [inline scripts/check_patch_checks.py:57]: This looks good as long as it's invoked with old ( > 6hr old) timestamps. 
Newer stuff won't have testsuite reports.
That way if github doesn't run the scheduled run 4 times in a row we process a patch twice... not ideal but that's fine.

File: scripts/check_patch_checks.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    testsuite_reported = False
+    
+    for check in checks:
+        if check["description"] == "Patch failed to apply":
+            apply_failure = True
+        if check["context"] == "toolchain-ci-rivos-test":
+            testsuite_report = True
+
+    # If testsuite was not reported, mark for rerun
+    # testsuite should be reported as long as it was applied
```
Comment (patrick-rivos) [inline scripts/check_patch_checks.py:49]: (& enough time has passed)

File: scripts/get_previous_timestamp.py [hunk 8]
```diff
@@ -35,24 +41,38 @@ def get_workflow_runs(token: str, repo: str):
...
-    runs = json.loads(r.text)
+    if r.status_code >= 500:
+        with open("patchwork_down.txt", "w") as f:
+            f.write(f"status code: {r.status_code}")
+        return None
+    runs = [run for run in json.loads(r.text) if run["name"] == workflow]
+    print(f"After filter have {len(runs)} to consider")
     return runs
 
 def get_run_timestamp(runs, run_id: str):
```
Comment (patrick-rivos) [inline scripts/get_previous_timestamp.py:52]:
```suggestion
def write_run_timestamp(runs, run_id: str):
```

File: scripts/get_previous_timestamp.py [hunk 9]
```diff
@@ -35,24 +41,38 @@ def get_workflow_runs(token: str, repo: str):
...
 
 def main():
     args = parse_arguments()
     runs = get_workflow_runs(args.token, args.repo)
+    if runs == None:
+        print("Server failure")
+        return
     with open("runs.log", "w") as f:
         f.write(json.dumps(runs['workflow_runs'][:10], indent=4))
     get_run_timestamp(runs['workflow_runs'], args.run_id)
```
Comment (patrick-rivos) [inline scripts/get_previous_timestamp.py:74]:
```suggestion
    write_run_timestamp(runs['workflow_runs'], args.run_id)
```

**Review by patrick-rivos — CHANGES_REQUESTED** [review]
lots of misc stuff/some concerns

File: scripts/get_previous_timestamp.py (hunk 4 above)
Comment (patrick-rivos) [reply scripts/get_previous_timestamp.py:42]: nvm, filter is above

File: scripts/check_patch_checks.py (hunk 6 above)
Comment (ewlu) [reply scripts/check_patch_checks.py:57]: I'm not sure if we should run this just one day behind or two days behind. If it's two days behind, we don't need to worry about duplicate patches but the latency increases.

File: scripts/get_previous_timestamp.py (hunk 1 above)
Comment (ewlu) [reply scripts/get_previous_timestamp.py:71]: I'll exit 1

**Review by patrick-rivos — APPROVED** [review]
LGTM

**Review by patrick-rivos — APPROVED** [review]

**Pull request merged**